site/src/contexts/auth/useAuth.test.tsx

```diff
@@ -0,0 +1,60 @@
+import { renderHook } from "@testing-library/react";
+import type { FC, PropsWithChildren } from "react";
+import { QueryClientProvider } from "react-query";
+import { createTestQueryClient } from "testHelpers/renderHelpers";
+import { AuthProvider } from "./AuthProvider";
+import { useAuth, useAuthenticated } from "./useAuth";
+
+const Wrapper: FC<PropsWithChildren> = ({ children }) => {
+  return (
+    <QueryClientProvider client={createTestQueryClient()}>
+      <AuthProvider>{children}</AuthProvider>
+    </QueryClientProvider>
+  );
+};
+
+describe("useAuth", () => {
+  it("throws an error if it is used outside of <AuthProvider />", () => {
+    jest.spyOn(console, "error").mockImplementation(() => {});
+
+    expect(() => {
+      renderHook(() => useAuth());
+    }).toThrow("useAuth should be used inside of <AuthProvider />");
+
+    jest.restoreAllMocks();
+  });
+
+  it("returns AuthContextValue when used inside of <AuthProvider />", () => {
+    renderHook(() => useAuth(), {
+      wrapper: Wrapper,
+    });
+  });
+});
+
+const RequireAuthWrapper: FC<PropsWithChildren> = ({ children }) => {
+  const { user } = useAuth();
+  return user ? <>{children}</> : null;
+};
+
+describe("useAuthenticated", () => {
```

Comment:
Going back to the race condition comment – do you feel like this is something that should be covered by a test case?

Reply:
Yes, it should!